packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -477,7 +447,11 @@ export default class RestApi {
         await sendObjects(
           Object.values(
             app.getObjects({
-              filter: every(_ => _.type === 'message' && _.$object === id, handleOptionalUserFilter(query.filter)),
+              filter: every(
+                messagesFilter({ excludeAlarms: true }),
+                _ => _.$object === id,
```

Comment:
since every check the filter in order, I think we should apply the most restrictive filter first, , and it is probably the $object one